src/libs/ajax.js

```diff
@@ -55,7 +55,7 @@ const withCancellation = wrappedFetch => async (...args) => {
   }
 }
 
-const withRetryOnError = _.curry(wrappedFetch => async (...args) => {
+const withRetryOnError = _.curry((shouldNotRetryFn, wrappedFetch) => async (...args) => {
```

Comment:
Another minor comment, in general I prefer positive over negative names, since if you ever have to negate them in an expression it makes it harder (for me) to figure out what's going on. So maybe `shouldThrowFn` or `shouldRetryFn` and negate the sense in the caller.